### DISCUSSION TITLE
add the condition with cb['fillout_factor']

### DISCUSSION THREAD

**Discussion by dingxu6207** [phoebe-project/phoebe2#481] [Questions]
when I run the example    ‘https://nbviewer.jupyter.org/github/phoebe-project/phoebe2-workshop/blob/2019july/Intro_Tutorial_16_contact_binaries.ipynb’

cb['q'] = 1.
cb['fillout_factor'] = 0.5  #this is error
cb.run_compute(irrad_method='none')

rise the error  ‘ValueError: cannot change the value of a constrained parameter.  This parameter is constrained by 'q@binary@component, pot@contact_envelope@component'

but the example of website run successfully, How can I solve it?

**Comment by dingxu6207** [comment]
I have solved it, thank you!

**Reply by kecnry** [reply]
Glad to hear you figured it out - let us know if you run into any other questions or issues!

Just as a heads up - the notebook you mentioned is from a workshop two years ago which are _not_ kept up-to-date with changes to phoebe, so you may run into the occasional problem.  Most of the same content is covered in the [online tutorials](http://phoebe-project.org/docs/latest/tutorials) which are kept up-to-date.  These in particular cover contact binaries:

* [Contact Binary Hierarchy](http://phoebe-project.org/docs/latest/tutorials/contact_binary_hierarchy.ipynb)
* [Critical Radii: Contact Systems (requiv_min & requiv_max)](http://phoebe-project.org/docs/latest/tutorials/requiv_crit_contact.ipynb)
* [Minimal Contact Binary System](http://phoebe-project.org/docs/latest/examples/minimal_contact_binary.ipynb)
* [Contact Binary with Spots](http://phoebe-project.org/docs/latest/examples/contact_spots.ipynb)
* [Comparing Contact Binary System PHOEBE 2 vs PHOEBE Legacy](http://phoebe-project.org/docs/latest/examples/legacy_contact_binary.ipynb)
